### ISSUE TITLE
ValueTask<TResult>.GetAwaiter().GetResult() is unnecessarily complex

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn-analyzers#3388]
It would be nice to have a new analyzer that detected cases of `ValueTask<TResult>.GetAwaiter().GetResult()` and offered to convert them to `ValueTask<TResult>.Result`. Unlike `Task<TResult>`, these operations have the same behavior for `ValueTask<TResult>`.

**Comment by Evangelink** [comment]
I see that quite some analyzers related to threading (e.g. `ConfigureAwait`) have been superseded by the vsthreading project. Isn't it better to also have this rule on the other repo?

**Comment by Evangelink** [comment]
@sharwell @AArnott Do we/you need to move this ticket and do the implementation of the vsthreading repo?

**Comment by AArnott** [comment]
@Evangelink For years vs-threading had the best threading/async analyzers, yes. And that's still true, but the winds are changing. The .NET Core SDK wants to define all of the threading/async analyzers that are of general interest (i.e. not specific to the vs-threading library). They're in the process of migrating analyzers now. 

I don't know if this is the right repo to track the .NET SDK analyzers. @terrajobst do you know?

**Comment by Evangelink** [comment]
So the various tickets I have open regarding threading analyzers will be moved too? Anyways thank you for the quick feedback!

**Comment by AArnott** [comment]
Yes, open issues on vs-threading requesting new analyzers won't be buried. We'll keep them active and/or move them to the right repo as this progresses.
